File: src/examples/closeSignatureOrder/basic.py
from criipto_signatures import CriiptoSignaturesSDKSync
from criipto_signatures.models import (
  CreateSignatureOrderInput,
  DocumentInput,
  PadesDocumentInput,
  DocumentStorageMode,
  CloseSignatureOrderInput,
  JWTSignature,
)
from criipto_signatures.operations import (
  CloseSignatureOrder_CloseSignatureOrderOutput_SignatureOrder_Document_PdfDocument,
)

client = CriiptoSignaturesSDKSync(
  "{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"
)

# Create signature order
signatureOrder = client.createSignatureOrder(
  CreateSignatureOrderInput(
    title="Python sample",
    documents=[
      DocumentInput(
        pdf=PadesDocumentInput(
          title="My document",
          blob=bytes("...", "utf-8"),  # bytes object, or a base64 encoded string
          storageMode=DocumentStorageMode.Temporary,
        )
      )
    ],
  )
)


# Close signature order
closedSignatureOrder = client.closeSignatureOrder(
  CloseSignatureOrderInput(signatureOrderId=signatureOrder.id)
)

for document in closedSignatureOrder.documents:
  if (
    isinstance(
      document,
      CloseSignatureOrder_CloseSignatureOrderOutput_SignatureOrder_Document_PdfDocument,
    )
  ) and document.signatures is not None:
    # document.blob is a bytes object containing the signed/closed PDF
    for signature in document.signatures:
      if isinstance(signature, JWTSignature):
        # signature.jwt holds the authentication evidence used in the signature
        print(signature.jwt)

File: src/examples/changeSignatureOrder/basic.cs
using Criipto.Signatures;
using Criipto.Signatures.Models;

using (var client = new CriiptoSignaturesClient("{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"))
{
    // Setup document input
    var documents = new List<DocumentInput>
    {
        new DocumentInput
        {
            pdf = new PadesDocumentInput
            {
                title = "Dotnet Sample",
                blob = new Byte[64], // Should be the bytes of a PDF file
                storageMode = DocumentStorageMode.Temporary
            }
        }
    };

    // Setup signature order input
    var createSignatureOrderInput = new CreateSignatureOrderInput
    {
        title = "Dotnet Sample",
        documents = documents,
        maxSignatories = 10
    };

    // Create signature order
    var signatureOrder = await client.CreateSignatureOrder(createSignatureOrderInput);

    // Change signature order to support 20 signatories
    await client.ChangeSignatureOrder(
        new ChangeSignatureOrderInput
        {
            signatureOrderId = signatureOrder.id,
            maxSignatories = 20
        }
    );
}
